pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/pubsub/MercuryOrderPublisher.java

```diff
@@ -98,44 +96,28 @@ public void onSuccess(String messageId) {
         }
     }
 
-    public void createAndPublishMessage(String[] barcodes, String projectId, String topicId, DDPInstance ddpInstance,
-                                        String ddpParticipantId) {
+    public void createAndPublishMessage(String[] barcodes, String projectId, String topicId, DDPInstanceDto ddpInstance,
+                                        String ddpParticipantId, String collaboratorParticipantId) {
+        if (requestHadCollaboratorId(ddpParticipantId, collaboratorParticipantId)) {
+            Optional<String> maybeParticipantId =
+                    participantDao.getParticipantFromCollaboratorParticipantId(collaboratorParticipantId,
+                            String.valueOf(ddpInstance.getDdpInstanceId()));
+            ddpParticipantId = maybeParticipantId.orElseThrow();
+        }
         log.info("Publishing message to mercury");
         String researchProject = ddpInstance.getResearchProject();
-        String creatorId = new InstanceSettings().getInstanceSettings(ddpInstance.getName()).getMercuryOrderCreator().orElseThrow();
+        String creatorId = ddpInstance.getMercuryOrderCreator().orElseThrow();
         String mercuryOrderId = createMercuryUniqueOrderId();
         MercuryPdoOrder mercuryPdoOrder = new MercuryPdoOrder(creatorId, mercuryOrderId, researchProject, barcodes);
         String json = new Gson().toJson(mercuryPdoOrder);
         if (StringUtils.isNotBlank(json)) {
-            orderOnMercury(projectId, topicId, json, barcodes, ddpParticipantId, mercuryOrderId);
+            this.mercuryOrderDao.orderOnMercury(projectId, topicId, json, barcodes, ddpParticipantId, mercuryOrderId, this);
```

Comment:
I don't like to give in `this` to `orderOnMercury` please refactor to have the dao method return what is needed and then order it from here!

Reply:
So if you look at the `OrderOnMercury` I am doing the insertion of db and the publish on pub sub in thee same method to be  able to roll back the db when sth goes wrong with pubsub.